addons/mail/static/src/components/notification_list/notification_list.js

```diff
@@ -135,10 +136,23 @@ class NotificationList extends Component {
                     };
                 }).concat(notifications);
         }
-        return {
+
+        const res = {
             isDeviceMobile: this.env.messaging.device.isMobile,
             notifications,
         };
+
+        if (
+            props.filter === 'all' &&
+                this.env.messaging.isNotificationPermissionDefault()
+        ) {
+            res.notifications.unshift({
+                type: 'odoobotRequest',
+                uniqueId: `odoobotRequest`,
+            });
+        }
```

Comment:
here could push "odoobot has a request" notification before adding thread notifications. `unshift()` made sense with override because notifications required super call.